hydra-node/src/Hydra/Ledger/Cardano/Configuration.hs

```diff
@@ -32,26 +34,43 @@ readJsonFileThrow parser filepath = do
 shelleyGenesisFromJson :: Json.Value -> Json.Parser (Ledger.ShelleyGenesis LedgerEra)
 shelleyGenesisFromJson = parseJSON
 
--- FIXME: We should not need the shelley genesis as the Ledger.Globals are
--- irrelevant (e.g. active slot coefficient) or can be derived from chain config
--- (e.g. networkId). Consequence of this would be less configuration for the
--- hydra-node (only protocol-parameters).
-newGlobals :: Ledger.ShelleyGenesis LedgerEra -> Ledger.Globals
-newGlobals shelleyGenesis =
-  Ledger.mkShelleyGlobals
-    shelleyGenesis
-    epochInfo
-    maxProtocolVersion
+data GlobalsTranslationError = GlobalsTranslationError deriving (Eq, Show)
+
+instance Exception GlobalsTranslationError
+
+-- | Create new L2 ledger 'Globals' from 'GenesisParameters'.
+-- NOTE: hydra-node will fail early if the conversion is not successfull.
+newGlobals :: MonadThrow m => GenesisParameters -> m Globals
+newGlobals genesisParameters = do
+  case boundRational @PositiveUnitInterval protocolParamActiveSlotsCoefficient of
+    Nothing -> throwIO GlobalsTranslationError
+    Just slotCoeff' -> do
+      let slotCoeff = mkActiveSlotCoeff slotCoeff'
```

Comment:
Could: use `fmap` to avoid binding two variables

```suggestion
  case mkActiveSlotCoeff <$> boundRational protocolParamActiveSlotsCoefficient of
    Nothing -> throwIO GlobalsTranslationError
    Just activeSlotCoeff -> do
```